lib/sass/script/value/number.rb

```diff
@@ -34,31 +34,37 @@ class Number < Base
     attr_accessor :original
 
     def self.precision
-      @precision ||= 5
+      Thread.current[:sass_numeric_precision] || Thread.main[:sass_numeric_precision] || 5
     end
 
     # Sets the number of digits of precision
     # For example, if this is `3`,
     # `3.1415926` will be printed as `3.142`.
+    # The numeric precision is stored as a thread local for thread safety reasons.
+    # To set for all threads, be sure to set the precision on the main thread.
     def self.precision=(digits)
-      @precision = digits.round
-      @precision_factor = 10.0**@precision
-      @epsilon = 1 / (@precision_factor * 10)
+      Thread.current[:sass_numeric_precision] = digits.round
+      Thread.current[:sass_numeric_precision_factor] = 10.0**precision
+      Thread.current[:sass_numeric_epsilon] = 1 / (precision_factor * 10)
```

Comment:
Rather than duplicating these here, it might be better to just let them be lazily initialized in the getters.